runner/src/lib.rs

```diff
@@ -66,15 +66,47 @@ fn run_module(module: &Module, entry_point: Option<&str>) -> Result<(), String>
         return Err("Target doesn't have a target machine.".to_owned());
     }
 
-    let entry_point = choose_entry_point(module_functions(module), entry_point)?;
     inkwell::support::load_library_permanently("");
+
     let execution_engine = module
         .create_jit_execution_engine(OptimizationLevel::None)
         .map_err(|e| e.to_string())?;
 
     bind_functions(module, &execution_engine)?;
 
-    unsafe { run_entry_point(&execution_engine, entry_point) }
+    let entry_point = choose_entry_point(module_functions(module), entry_point)?;
+    // TODO: need a cleaner way to get the attr strings for metadata
+    let attrs: Vec<(String, String)> = entry_point
+        .attributes(AttributeLoc::Function)
+        .iter()
+        .map(|attr| {
+            (
+                attr.get_string_kind_id()
+                    .to_str()
+                    .expect("Invalid UTF8 data")
+                    .to_string(),
+                attr.get_string_value()
+                    .to_str()
+                    .expect("Invalid UTF8 data")
+                    .to_string(),
+            )
+        })
+        .collect();
+
+    for _ in 1..=1 {
```

Comment:
Is this a placeholder for shots? If so, it would be good to add a comment indicating that and to make it easier for future devs to find.